docs/frontend/forms.md

```diff
@@ -0,0 +1,326 @@
+```vue
+<template>
+  <T3Form
+    ref="form"
+    :elements="form.elements"
+    :classes="css"
+    @submit="onSubmit"
+  >
+    <template #before="{model}">
+      {{ model }}
+    </template>
+
+    <template #cta>
+      <button type="submit">
+        Send form
+      </button>
+    </template>
+
+    <template #after="{state}">
+      <p v-if="state.error || state.failture">
+        {{ i18n.serverError }}
+      </p>
+      <p v-if="state.success">
+        {{ i18n.serverSuccess }}
+      </p>
+    </template>
+  </T3Form>
+</template>
+<script>
+import T3CeFormFormframework from '~typo3/components/T3CeFormFormframework/T3CeFormFormframework.vue'
+export default {
+  extends: T3CeFormFormframework,
+   data () {
+    return {
+      css: Object.freeze({
+        // field.identifier : class name
+        name: 'wrap-my-name-field'
+      })
+    }
+  },
+  methods: {
+    onSuccess (actionAfterSuccess) {
+      if (actionAfterSuccess?.redirectUri) {
+        this.$router.push(actionAfterSuccess.redirectUri)
+      } else {
+        // do something else
+      }
+    },
+  }
+}
+</script>
+```
+
+## Customize field templates
+
+Each form field use [`T3FormField`](https://github.com/TYPO3-Initiatives/nuxt-typo3/blob/master/lib/components/T3Form/T3FormField/T3FormField.vue) component as the base template. You can override this template:
+
+Create and register global components `components/T3FormField.vue`:
+```vue
+<template>
+  <ValidationProvider
+    v-slot="state"
+    :vid="field.identifier"
+    :name="field.label"
+    :rules="rules"
+    slim
+  >
+    <div class="field-row">
+      <label :for="field.identifier"> {{ field.label }} </label>
+      <slot :state="state">
+        <input
+          :id="field.identifier"
+          v-model="innerValue"
+          :type="field.type"
+          :label="field.label"
+          :name="field.name || field.identifier"
+          :class="state.classes"
+          :placeholder="placeholder"
+          :required="required"
+          @input="(event) => $emit('input', event.target.value)"
+        >
+      </slot>
+      <ul v-if="state.errors.length" class="t3-form-field__errors">
+        <li v-for="(error, key) in state.errors" :key="key">
+          {{ error }}
+        </li>
+      </ul>
+    </div>
+  </ValidationProvider>
+</template>
+<script>
+import { T3FormField } from '~typo3/components/T3Form'
+export default {
+  name: 'T3FormField',
+  extends: T3FormField
+}
+</script>
+<style scoped>
+.field-row {
+  display: flex;
+  align-items: center;
+  margin: 10px 0;
+}
+
+label {
+  font-weight: bold;
+  margin-right: 10px;
+}
+
+ul {
+  margin: 0;
+  color: red;
+}
+</style>
+```
+This component use [`<ValidationProvider/>`](https://vee-validate.logaretm.com/v3/api/validation-provider.html) for field validation.
+This template will be suitable for most regular input elements. You can use this template to provide more form types like textarea, checkbox.
```

Comment:
You can use this template to provide more form elements like textarea, checkbox, ect.